apps/daimo-mobile/src/view/screen/send/CoinDisplay.tsx

```diff
@@ -1,113 +1,211 @@
-import { ForeignToken, baseUSDC, supportedSendCoins } from "@daimo/common";
-import { View, StyleSheet, Text, Pressable } from "react-native";
+import {
+  DAv2Chain,
+  ForeignToken,
+  arbitrum,
+  arbitrumSepolia,
+  arbitrumUSDC,
+  base,
+  baseSepolia,
+  baseUSDC,
+  getChainDisplayName,
+  optimism,
+  optimismSepolia,
+  optimismUSDC,
+  polygon,
+  polygonAmoy,
+  polygonUSDC,
+} from "@daimo/common";
+import {
+  Image,
+  ImageSourcePropType,
+  Pressable,
+  StyleSheet,
+  Text,
+  View,
+} from "react-native";
 import SelectDropdown from "react-native-select-dropdown";
 
+import IconArbitrum from "../../../../assets/logos/arb-logo.png";
+import IconBase from "../../../../assets/logos/base-logo.png";
+import IconOptimism from "../../../../assets/logos/op-logo.png";
+import IconPolygon from "../../../../assets/logos/poly-logo.png";
 import { useAccount } from "../../../logic/accountManager";
-import { DropdownPickButton } from "../../shared/DropdownPickButton";
 import { color, ss } from "../../shared/style";
-import { DaimoText } from "../../shared/text";
+
+type SendPair = {
+  chain: DAv2Chain;
+  coin: ForeignToken;
+};
+
+// Get the logo for the chain
+const getChainUri = (chain: DAv2Chain) => {
+  switch (chain) {
+    case base:
+    case baseSepolia:
+    default:
+      return IconBase;
+    case optimism:
+    case optimismSepolia:
+      return IconOptimism;
+    case polygon:
+    case polygonAmoy:
+      return IconPolygon;
+    case arbitrum:
+    case arbitrumSepolia:
+      return IconArbitrum;
+  }
+};
```

Comment:
maybe move this to daimo common in the future

Reply:
Reason is loading from mobile bundle (../../../../assets/...) vs. loading from fetched source url 

I think the former is faster? but wdyt